src/gp_engine.hpp

```diff
@@ -241,13 +268,14 @@ class GPEngine {
   // space).
   static constexpr double max_log_branch_length_ = 1.1;
   // Precision used for checking convergence of branch length optimization.
-  int significant_digits_for_optimization_ = 6;
-  //
-  double relative_tolerance_for_optimization_ = 1e-2;
-  // Step size used for gradient-based branch length optimization.
+  int significant_digits_for_optimization_ = 10;
```

Comment:
Wasn't there something surprising about the way this was set up? Like it was some number of significant digits in the binary representation or ...? A comment would help.

Reply:
Added some text describing this